diff --git a/node_modules/colors/lib/colors.js b/node_modules/colors/lib/colors.js
index 59898de..f7c72c1 100644
--- a/node_modules/colors/lib/colors.js
+++ b/node_modules/colors/lib/colors.js
@@ -122,16 +122,7 @@ function applyTheme (theme) {
 }
 
 colors.setTheme = function (theme) {
-  if (typeof theme === 'string') {
-    try {
-      colors.themes[theme] = require(theme);
-      applyTheme(colors.themes[theme]);
-      return colors.themes[theme];
-    } catch (err) {
-      console.log(err);
-      return err;
-    }
-  } else {
+  if (typeof theme !== 'string') {
     applyTheme(theme);
   }
 };
